### PR TITLE
Added mini-batch support to GaussianNB

### PR DESCRIPTION
**Pull request by BALOGUN-DAVID**
## Summary

- Add mini-batch support to `naive_bayes.GaussianNB`
- Implement batch learning and batch joint log-likelihood computation
- Add tests covering dense, sparse, missing-feature, and unfitted mini-batch behavior

## Tests

- `uv run pytest river/naive_bayes/test_naive_bayes.py`
- `uv run pytest river/proba/test_gaussian.py`
- `uv run pytest river/naive_bayes/test_naive_bayes.py river/proba/test_gaussian.py river/test_estimators.py -k "GaussianNB or test_univariate_multivariate_consistency"`
- `uv run pre-commit run --files river/naive_bayes/gaussian.py river/naive_bayes/test_naive_bayes.py river/proba/gaussian.py docs/releases/unreleased.md`

### CODE REVIEW

File: river/naive_bayes/gaussian.py [hunk 1]
```diff
@@ -41,6 +46,16 @@ class GaussianNB(base.Classifier):
...
 
+    You can also train the model and make predictions in mini-batch mode.
+
+    >>> import pandas as pd
+
+    >>> model = naive_bayes.GaussianNB()
+    >>> model.learn_many(pd.DataFrame(X), pd.Series(Y))
+    >>> model.predict_many(pd.DataFrame([[-0.8, -1]]))
+    0    1
+    dtype: int64
```
Comment (MaxHalford) [inline river/naive_bayes/gaussian.py]: It's more relevant to provide an example with multiple rows

**Review by MaxHalford — COMMENTED** [review]
Nice! I would like to see some tests that compare to sklearn's [GaussianNB](https://scikit-learn.org/stable/modules/generated/sklearn.naive_bayes.GaussianNB.html), which has a `partial_fit` method

**Comment by BALOGUN-DAVID** [comment]
Thanks for the suggestion. I added a test that trains River's `GaussianNB.learn_many` and sklearn's `GaussianNB.partial_fit` on matching mini-batches, then compares class means, class priors, predictions, and predicted probabilities.

File: river/naive_bayes/test_naive_bayes.py [hunk 2]
```diff
@@ -238,3 +239,122 @@ def test_river_vs_sklearn(model, sk_model, bag):
...
+    classes = np.array(["a", "b"])
+
+    model = naive_bayes.GaussianNB()
+    sk_model = sk_naive_bayes.GaussianNB(var_smoothing=0.0)
+
+    for start in range(0, len(X), 50):
+        X_batch = X.iloc[start : start + 50]
+        y_batch = y.iloc[start : start + 50]
+        model.learn_many(X_batch, y_batch)
+        sk_model.partial_fit(X_batch, y_batch, classes=classes)
```
Comment (MaxHalford) [inline river/naive_bayes/test_naive_bayes.py:342]: I'd rather you move the predict tests here, inside the loop, instead of after the loop. That way we test more edge cases.

**Comment by BALOGUN-DAVID** [comment]
Updated as suggested: the sklearn prediction checks now run after each mini-batch update.
